bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/VoiceManagerImpl.java

```diff
@@ -291,19 +291,22 @@ public void say(String text, @Nullable String voiceId, @Nullable String sinkId,
                             e.getMessage(), e);
                 }
             }
-            try {
-                sink.process(audioStream);
-            } finally {
-                if (volume != null && oldVolume != null) {
+
+            final PercentType oldVolumeFinal = oldVolume;
+            Runnable toRunWhenProcessFinished = () -> {
+                if (volume != null && oldVolumeFinal != null) {
                     // restore volume only if it was set before
                     try {
-                        sink.setVolume(oldVolume);
+                        sink.setVolume(oldVolumeFinal);
```

Comment:
Some audio sinks will throw `UnsupportedOperationException`, for example the freebox audio sink.
Maybe this exception should be caught and ignored ?

Reply:
Seems legit. Done.